lib/lightning_web/live/user_live/form_component.ex

```diff
@@ -31,28 +31,42 @@ defmodule LightningWeb.UserLive.FormComponent do
   end
 
   defp save_user(socket, :edit, user_params) do
-    case Accounts.update_user_details(socket.assigns.user, user_params) do
-      {:ok, _user} ->
-        {:noreply,
-         socket
-         |> put_flash(:info, "User updated successfully")
-         |> push_redirect(to: socket.assigns.return_to)}
-
-      {:error, %Ecto.Changeset{} = changeset} ->
-        {:noreply, assign(socket, :changeset, changeset)}
+    if socket.assigns.can_edit_users do
```

Comment:
Is this working ? Shouldn't you reassign the socket in the `def update/2` function ?